backend/papersfeed/tests/tests_subscription.py

```diff
@@ -0,0 +1,177 @@
+                               json.dumps({
+                                   constants.ID: follow_id
+                               }),
+                               content_type='application/json')
+
+        # User2 Sign out
+        client.delete('/api/session')
+
+        # User1 Sign in
+        client.get('/api/session',
+                   data={
+                       constants.EMAIL: 'user1@snu.ac.kr',
+                       constants.PASSWORD: 'iluvswpp1234'
+                   },
+                   content_type='application/json')
+
+        # User1 Likes paper1
+        paper_id = Paper.objects.filter(title='paper1').first().id
+        client.post('/api/like/paper',
+                    data=json.dumps({
+                        constants.ID: paper_id,
+                    }),
+                    content_type='application/json')
+        # User1 created collection1
+        client.post('/api/collection',
+                    json.dumps({
+                        constants.TITLE: 'SWPP Papers Test',
+                        constants.TEXT: 'papers for swpp 2019 class Test'
+                    }),
+                    content_type='application/json')
+
+        # User1 Likes review1
+        review_id = Review.objects.filter(title='review1').first().id
+        client.post('/api/like/review',
+                    data=json.dumps({
+                        constants.ID: review_id,
+                    }),
+                    content_type='application/json')
+
+        # User1 Sign out
+        client.delete('/api/session')
+
+        # User2 Sign In
+        client.get('/api/session',
+                   data={
+                       constants.EMAIL: 'user2@snu.ac.kr',
+                       constants.PASSWORD: 'iluvswpp1234'
+                   },
+                   content_type='application/json')
+        # User2 Get Subscriptions
+        response = client.get('/api/subscription',
+                              data={
+                                  constants.PAGE_NUMBER: 1
+                              },
+                              content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content.decode())[constants.IS_FINISHED], True)
+        self.assertEqual(int(json.loads(response.content.decode())[constants.PAGE_NUMBER]), 1)
+
+        user1_id = User.objects.get(email='user1@snu.ac.kr').id
+
+        # there should be one subscription item
+        subscriptions = json.loads(response.content)['subscriptions']
+        likes_paper = subscriptions[0]
+        created_collection = subscriptions[1]
+        likes_review = subscriptions[2]
+        # test for likes_paper
+        self.assertEqual(likes_paper['actor'], {
+            constants.ID: user1_id,
+            constants.USERNAME: 'user1',
+        })
+        self.assertEqual(likes_paper['verb'], 'liked')
+        # paper = Paper.objects.get(id=paper_id)
```

Comment:
Why did you comment this line?? I think this is important.

Reply:
The solution for testing 'action_object' is not that simple (or as I thought), since they are the result of 'get_papers', 'get_collections' or 'get_reviews' from each of the 'utils' they are related of. I addtionaly left "FIXME" comment for the problem.